pkg/registry/apis/folders/sub_access.go

```diff
@@ -45,35 +47,78 @@ func (r *subAccessREST) NewConnectOptions() (runtime.Object, bool, string) {
 }
 
 func (r *subAccessREST) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
-	ns, err := request.NamespaceInfoFrom(ctx, true)
+	// need to get the object to get the folder
+	obj, err := r.getter.Get(ctx, name, &v1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
-	user, err := identity.GetRequester(ctx)
-	if err != nil {
-		return nil, err
+
+	folder, ok := obj.(*v0alpha1.Folder)
+	if !ok {
+		return nil, fmt.Errorf("expected a folder object")
 	}
 
-	// Can view is managed here (and in the Authorizer)
-	f, err := r.service.Get(ctx, &folder.GetFolderQuery{
-		UID:          &name,
-		OrgID:        ns.OrgID,
-		SignedInUser: user,
-	})
+	m, err := utils.MetaAccessor(folder)
 	if err != nil {
 		return nil, err
 	}
-	guardian, err := guardian.NewByFolder(ctx, f, ns.OrgID, user)
-	if err != nil {
-		return nil, err
+
+	// Check for the request claims
+	user, ok := claims.From(ctx)
+	if !ok {
+		return nil, errNoUser
 	}
 
-	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+	// TODO! this client should have a better bulk request mode
```

Comment:
Yes I have this one in my todos, https://github.com/grafana/identity-access-team/issues/1103